client/server/middleware/test/geoip-header.test.ts

```diff
@@ -0,0 +1,98 @@
+import { type Request, type Response } from 'express';
+import middlewareGeoipHeader from '../geoip-header';
+
+const mockInfo = jest.fn();
+
+jest.mock( 'calypso/server/lib/logger', () => ( {
+	getLogger: jest.fn().mockImplementation( () => ( { info: mockInfo } ) ),
+} ) );
+
+describe( 'geoip-header middleware', () => {
+	let request: Request;
+	let response: Response;
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+		jest.spyOn( global, 'fetch' );
+		request = { headers: {} } as Request;
+		response = {} as Response;
+	} );
+
+	it( 'should fetch country code on initialization', async () => {
+		const { promise, resolve } = Promise.withResolvers< void >();
+		( global.fetch as jest.Mock ).mockImplementation( () => {
+			resolve();
+			return Promise.resolve( {
+				json: () => Promise.resolve( { country_short: 'US' } ),
+			} );
+		} );
```

Comment:
This could be a good example for `async` without `await` 🙂 I'm not asking to change it, just saying that it looks pretty good:
```js
mockImplementation( async () => {
  resolve();
  return {
    json: async () => ( { country_short: 'US' } ),
  };
} );
```